From e54f5f3dc8ec281bbc17e4092d168ea32ae099d7 Mon Sep 17 00:00:00 2001
From: Max Romanov <max.romanov@nginx.com>
Date: Tue, 22 Oct 2019 16:04:30 +0300
Subject: [PATCH] Python: fixing build for Python 3.8.

Thanks to tonyafanasyev.
This is related to #331 issue on GitHub.
---
 auto/modules/python   | 4 ++++
 src/nxt_python_wsgi.c | 4 ++++
 2 files changed, 8 insertions(+)

diff --git a/auto/modules/python b/auto/modules/python
index abd145c9..ecf8802a 100644
--- a/auto/modules/python
+++ b/auto/modules/python
@@ -64,6 +64,10 @@ nxt_found=no
 
 if /bin/sh -c "$NXT_PYTHON_CONFIG --prefix" >> $NXT_AUTOCONF_ERR 2>&1; then
 
+    if ${NXT_PYTHON_CONFIG} --embed 2>/dev/null; then
+        NXT_PYTHON_CONFIG="${NXT_PYTHON_CONFIG} --embed"
+    fi
+
     NXT_PYTHON_INCLUDE=`${NXT_PYTHON_CONFIG} --includes`
     NXT_PYTHON_LIBS=`${NXT_PYTHON_CONFIG} --ldflags`
 
diff --git a/src/nxt_python_wsgi.c b/src/nxt_python_wsgi.c
index a6d5f217..977c2998 100644
--- a/src/nxt_python_wsgi.c
+++ b/src/nxt_python_wsgi.c
@@ -179,6 +179,10 @@ static PyTypeObject nxt_py_input_type = {
 #if PY_MAJOR_VERSION == 3 && PY_MINOR_VERSION > 3
     0,                                  /* tp_finalize          */
 #endif
+#if PY_MAJOR_VERSION == 3 && PY_MINOR_VERSION >= 8
+    0,                                  /* tp_vectorcall        */
+    0,                                  /* tp_print             */
+#endif
 };
 
 
